runtime/autoload/man.vim

```diff
@@ -213,7 +213,11 @@ endfunction
 
 function! s:get_path(sect, name) abort
   if empty(a:sect)
-    return s:system(['man', s:find_arg, a:name])
+    " Some man implementations (OpenBSD) return all available man paths
+    " from the search command, so we split here to make sure we only
+    " take the first one.  Discovered and reported in
+    " https://github.com/neovim/neovim/issues/8341
+    return split(s:system(['man', s:find_arg, a:name]))[0]
```

Comment:
I guess split() could be empty, so should use `get()` instead of `[0]`:

      return get(split(s:system(['man', s:find_arg, a:name])), 0, '')

Also as @nhooyr mentioned the newline trim should be done here. So:

    return substitute(get(split(s:system(['man', s:find_arg, a:name])), 0, ''), '\n\+$', '', '')

Reply:
You'll need to remove the slice here that takes out the newline: https://github.com/jbernard/neovim/blob/a5dcfa1b7e07069ca5bcd0e9840adaa0e4138353/runtime/autoload/man.vim#L22

you'll also need to trim the newline in the second branch of `s:get_path`: https://github.com/jbernard/neovim/blob/a5dcfa1b7e07069ca5bcd0e9840adaa0e4138353/runtime/autoload/man.vim#L227